### PR TITLE
added endpoint regex for capabilityCheck metric

### PR DESCRIPTION
**Pull request by kristinaspring**
builds off of https://github.com/xmidt-org/webpa-common/pull/459

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -402,7 +404,16 @@ func authenticationHandler(v *viper.Viper, logger log.Logger, registry xmetrics.
 	var capabilityCheck CapabilityConfig
 	v.UnmarshalKey("capabilityCheck", &capabilityCheck)
 	if capabilityCheck.Type == "enforce" || capabilityCheck.Type == "monitor" {
-		checker, err := basculechecks.NewCapabilityChecker(capabilityCheckMeasures, capabilityCheck.Prefix, capabilityCheck.AcceptAllMethod)
+		var endpoints []*regexp.Regexp
+		for _, e := range capabilityCheck.EndpointBuckets {
+			r, err := regexp.Compile(e)
+			if err != nil {
+				logging.Warn(logger).Log(logging.MessageKey(), "failed to compile regular expression", "regex", e, logging.ErrorKey(), err.Error())
```
Comment (joe94) [inline main.go]: do we want to continue starting tr1d1um even when some endpoints were not loaded?

File: tr1d1um.yaml [hunk 2]
```diff
@@ -217,9 +217,19 @@ jwtValidator:
...
 #   type: "enforce"
+#   # prefix provides the regex to match the capability before the endpoint.
 #   prefix: "prefix Here"
+#   # acceptAllMethod provides a way to have a capability that allows all 
+#   # methods for a specific endpoint.
 #   acceptAllMethod: "all"
+#   # endpointBuckets provides regular expressions to use against the request 
+#   # endpoint in order to group requests for a metric label.
+#   endpointBuckets:
+#     - "hook"
```
Comment (joe94) [inline tr1d1um.yaml]: in our prod deployment, will we have an additional `hooks` endpoint?

File: main.go (hunk 1 above)
Comment (kristinaspring) [reply main.go]: I think so - a metric not configured quite correctly won't stop tr1d1um from doing its main job.  What do you think? I'm open to erroring out; I can also change the log to error, to make sure it gets picked up.

File: tr1d1um.yaml (hunk 2 above)
Comment (kristinaspring) [reply tr1d1um.yaml]: yeah I'll add that; thanks!

**Pull request merged**